qiskit_optimization/algorithms/minimum_eigen_optimizer.py

```diff
@@ -137,11 +151,18 @@ def __init__(
             TypeError: When one of converters has an invalid type.
             QiskitOptimizationError: When the minimum eigensolver does not return an eigenstate.
         """
-
+        if isinstance(min_eigen_solver, LegacyMinimumEigensolver):
+            warn_deprecated(
+                "0.5.0",
+                DeprecatedType.ARGUMENT,
+                f"min_eigen_solver as {LegacyMinimumEigensolver.__name__}",
+                new_name=f"min_eigen_solver as {SamplingMinimumEigensolver.__name__} "
+                f"or {NumPyMinimumEigensolver.__name__}",
+            )
         if not min_eigen_solver.supports_aux_operators():
             raise QiskitOptimizationError(
                 "Given MinimumEigensolver does not return the eigenstate "
-                + "and is not supported by the MinimumEigenOptimizer."
+                "and is not supported by the MinimumEigenOptimizer."
             )
```

Comment:
The supports_aux_operators used to imply that the algorithm had the state to measure aux ops. With the estimator this is no longer the case such that while new VQE can support_aux_operators it no longer has a eigenstate field. Nowdays if you want the state you would have to use the ansatz and optimal parameters from the result to compute it yourself, if that applies, say with a Sampler.

Reply:
Yes, `MinimumEigenOptimizer` requires sampler-based solver `SamplingVQE` instead of `VQE` due to this reason.